spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -381,7 +381,7 @@ private Schema calculateMetadataSchema(List<Types.NestedField> metaColumnFields)
     AtomicInteger nextId = new AtomicInteger();
     return new Schema(
         metaColumnFields,
-        table.schema().identifierFieldIds(),
+        Set.of(),
```

Comment:
I think I agree we probably should not be passing through the identifier fields here but the new test added still passes even without this fix. Are we sure this is the right place or is there another cause? Or perhaps there are more specific conditions.

Reply:
thanks for looking, I think i removed a lot of my original test due to the earlier review comment, let me double check how to reproduce it.

Reply:
Got it, i think it needs to query the metadata column.  I removed it as part of addressing Manu's comment.

Reply:
Ah I missed that too yeah the projection needs to include a metadata column, cool!